sabnzbd/downloader.py

```diff
@@ -436,8 +436,24 @@ def run(self):
         # Kick BPS-Meter to check quota
         sabnzbd.BPSMeter.update()
 
+        # Store when each server last searched for articles
+        last_searched = {}
+
+        # Store when each server last downloaded anything or found an article
+        last_busy = {}
+
         while 1:
             for server in self.servers:
+                serverid = server.id
+                if server.busy_threads:
+                    last_busy[serverid] = time.time()
```

Comment:
Sounds like a good approach.
Just looking through this new code it seems we call `time.time()` a lot, maybe we should only call it once per main-`while` loop and store it in a variable?
Not just for your new code, but also the rest of the code in the main loop.

Reply:
I decided to test it. It turns out `time.time()` is incredibly fast, so I don't think using a variable is necessary. On my Rpi 3, doing
`x = time.time()`
1M times took 2.19s, while
`x += 1`
took 1.33s. The difference is even smaller with Windows/Intel.

Reply:
Actually, maybe it would be a bit more useful than I first thought. When I subtract the time the loop itself takes and remove the assignment then calling time.time() takes about 5 times longer than calling a variable containing a float.

Reply:
👍